### PR TITLE
Add Dockerfile for Databricks runtime

### PR DESCRIPTION
**Pull request by wjxiz1992**
This Dockerfile is used for [Customize containers with Databricks Container Services](https://docs.databricks.com/clusters/custom-containers.html). 

To use it, user need to:
1. build the docker image locally
2. push the docker image to docker hub
3. set docker image url in Databricks cluster setup page.

Signed-off-by: Allen Xu <allxu@nvidia.com>

### CODE REVIEW

**Comment by tgravescs** [comment]
why do we need docker file?  Are there docs that point to this docker file?

File: examples/Spark-DL/Dockerfile.conda_db [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
```
Comment (sameerz) [inline examples/Spark-DL/Dockerfile.conda_db:2]:
```suggestion
 * Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
```

File: examples/Spark-DL/Dockerfile.conda_db [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    conda clean --all
+
+RUN pip uninstall tensorflow -y; pip install tf-nightly==2.7.0.dev20210722
+RUN pip uninstall keras-nightly -y; pip install keras-nightly==2.7.0.dev2021072200
+
+RUN mkdir -p /usr/local/nvidia/lib64 && \
+    ln -s /usr/local/cuda/lib64/libcusolver.so /usr/local/nvidia/lib64/libcusolver.so.10
+
+# install openjdk8, cmake, openmpi openmpi-mpicc
+RUN conda install cmake=3.19.6 openmpi openmpi-mpicc -y
```
Comment (sameerz) [inline examples/Spark-DL/Dockerfile.conda_db:60]: Is there any harm in specifying a conda channel, just to be specific?

File: examples/Spark-DL/Dockerfile.conda_db [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+RUN conda install cmake=3.19.6 openmpi openmpi-mpicc -y 
+
+RUN HOROVOD_WITH_MPI=1 HOROVOD_WITH_TENSORFLOW=1 HOROVOD_GPU_OPERATIONS=NCCL \
+    pip install horovod[spark] --no-cache-dir
+
+RUN conda install -c nvidia -c rapidsai -c numba -c conda-forge nvtabular=0.6.1 python=3.8 cudatoolkit=11.2
+
+RUN pip install pynvml
+
+RUN conda install ipython==7.19.0 matplotlib==3.4.2 jinja2==2.11.3
```
Comment (sameerz) [inline examples/Spark-DL/Dockerfile.conda_db:69]: Same question here, any harm in specifying a conda channel?

File: examples/Spark-DL/Dockerfile.conda_db (hunk 1 above)
Comment (wjxiz1992) [reply examples/Spark-DL/Dockerfile.conda_db]: Updated.

File: examples/Spark-DL/Dockerfile.conda_db (hunk 2 above)
Comment (wjxiz1992) [reply examples/Spark-DL/Dockerfile.conda_db]: Updated.

**Comment by wjxiz1992** [comment]
@tgravescs This is origined from an SA, asking if we can run this in DB runtime, I've verified in DB runtime.
I added some decription in README to mention this Dockerfile.

File: examples/Spark-DL/criteo_train/README.md [hunk 4]
```diff
@@ -85,3 +85,13 @@ If you want to try in a node with only 1 GPU, please modify the GPU number per w
 ## workspace folder in docker container:
 
 `/workspace/`
+
+## Run in Databriks Runtime
```
Comment (tgravescs) [inline examples/Spark-DL/criteo_train/README.md:89]: Databricks misspelled

Comment (wjxiz1992) [reply examples/Spark-DL/criteo_train/README.md]: Thanks! fixed.

**Review by nvliyuan — APPROVED** [review]

**Comment by wjxiz1992** [comment]
@nvliyuan let's merge this.

**Comment by nvliyuan** [comment]
> @nvliyuan let's merge this.

merged

**Pull request merged**